transcom/mymove
# INT B-19979 Diversion Reason in Prime API and UI

## [B-19979](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-19979)

## Summary

Adds the diversion reason field added in B-19938 and B-19939 to the Prime API and UI.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Access the office app
2. Login as a TOO
3. Find a move and shipment to request a diversion on, I used #AMDORD.
4. Navigate to the move task order page and locate the "Request diversion" button.
5. Submit a diversion request and provide a reason for your request.
6. Once the request has gone through successfully sign out and log into the Prime Simulator
7. Locate the move containing the shipment you requested a diversion on and open it
8. Ensure the diversion reason field exists and contains the reason you provided in step 5.
9. Optionally verify the diversion reason is also available in the [Prime V2](http://officelocal:3000/swagger-ui/prime_v2.html#/moveTaskOrder/getMoveTaskOrder) and [Prime V1](http://officelocal:3000/swagger-ui/prime.html#/moveTaskOrder/getMoveTaskOrder) APIs

### Frontend


### Backend


### Database

#### Any new migrations/schema changes:


## Screenshots
<img width="902" alt="Screenshot 2024-06-04 at 9 32 00 PM" src="https://github.com/transcom/mymove/assets/158500142/1d311a53-dce9-4d05-aff0-06abf5b20287">
